### ISSUE TITLE
game_loop and winit_input_helper do not appear to be compatible (invaders example)

### ISSUE THREAD

**Issue by Zageron** [parasyte/pixels#270]
I was using your invaders example as a reference and ran into an issue.
This isn't something you can fix, but thought I'd let you know.

`g.game.input.update(&event)` never returns true, which means the state can never be guaranteed to be cleared.

Ex. This will never run the keyboard events, and if I remove the if guard the keyboard state remains between events and I get multiple false positives.

The only state you care about is the example is key_pressed and window_resized, for actions that don't care about repeat triggers. Add a print statement to see what I mean.

```rust
game_loop(
    event_loop,
    window,
    game,
    FPS,
    0.1,
    |g| {
        g.game.update();
    },
    |g| {
        // Drawing
        g.game.world.draw(g.game.pixels.get_frame());
        if let Err(e) = g.game.pixels.render() {
            error!("pixels.render() failed: {}", e);
            g.exit();
        }

        #[cfg(not(target_arch = "wasm32"))]
        {
            // Sleep the main thread to limit drawing to the fixed time step.
            // See: https://github.com/parasyte/pixels/issues/174
            let dt = TIME_STEP.as_secs_f64() - Time::now().sub(&g.current_instant());
            if dt > 0.0 {
                std::thread::sleep(Duration::from_secs_f64(dt));
            }
        }
    },
    |g, event| {
        if g.game.input.update(&event) {
            println!("{:?}", event);
            // Close events
            if g.game.input.key_pressed(VirtualKeyCode::Escape) || g.game.input.quit() {
                g.exit();
            }

            if g.game.input.key_released(VirtualKeyCode::Space) {
                g.game.head.play_song();
                println!("Playing song");
            }

            // Resize the window
            if let Some(size) = g.game.input.window_resized() {
                g.game.pixels.resize_surface(size.width, size.height);
            }
        }
    },
);
```

**Comment by parasyte** [comment]
Looks like doing your own input accumulation is better in the short term, anyway. The current behavior of `winit_input_helper`'s `key_pressed` method isn't as documented: https://github.com/rukai/winit_input_helper/issues/17

**Comment by Zageron** [comment]
Wow thanks for the insanely thorough investigation and quick fix attempt. :o

**Comment by Zageron** [comment]
I tested our your branch as a dependency and it resolved my issues, thank you! :)
Once I get more progress done on my MVP I'll likely take into consideration your recommendation to refactor out `winit_input_helper` and write my own control logic.

**Comment by tuzz** [comment]
Hi, just thought I'd chip in and say I've released [version 0.9.0](https://crates.io/crates/game-loop) of game-loop that, once integrated, should resolve this problem. Cheers.
